test/test_s3.py

```diff
@@ -88,6 +92,11 @@ def setUp(self):
 
         self.put_body_stream = None
 
+        self.files = FileCreator()
+
+    def tearDown(self):
+        self.files.remove_all()
```

Comment:
need to call `super().tearDown()` afterwards
`NativeResourceTest.tearDown()` is where we check for leaks